molecule/swarm/playbook.yml

```diff
@@ -4,20 +4,48 @@
   roles:
     - role: ansible-traefik-docker
   tasks:
+    - name: Get docker networks
+      command: docker network ls
+      register: __networks
+    - name: Show networks
+      debug:
+        var: __networks.stdout_lines
     - name: start test web server
-      docker_container:
+      docker_swarm_service:
         name: testhttp
         image: deis/example-go
-        state: started
         networks:
-          - name: "{{ traefik_docker_network }}"
+          - "{{ traefik_docker_network }}"
         labels:
           test: webserver
           traefik.enable: "true"
+          traefik.port: "8080"
           traefik.docker.network: "{{ traefik_docker_network }}"
+    - name: Wait for traefik/consul to come up
+      wait_for:
+        port: 80
+    - name: Restart traefik
+      command:
+        cmd: docker service update --force traefik_traefik
+    - name: Try connecting
+      command:
+        cmd: curl -L -k -v -s -H "Host:traefik.traefik-swarm.docker.local" https://127.0.0.1:443/api/providers/docker
+      ignore_errors: true
+      failed_when: false
+      tags:
+        - skip_ansible_lint
+    - name: Restart traefik
+      command:
+        cmd: docker service update --force traefik_traefik
+    - name: Wait for traefik/consul to come up
+      wait_for:
+        port: 80
```

Comment:
So this works. It's also horrible. From hacking around it seems one needs to do one request and then restart traefik before everything works properly. I have no idea why. Sometimes, just a restart is enough - but what never seems to work is just letting it all come up cleanly from scratch.

*Maybe* things would work better if traefik would actually wait until the consul is running and configured before it starts up, but not 100% sure if that is the core of the issue.